milestones/tests/test_api.py

```diff
@@ -490,6 +518,72 @@ def test_add_course_content_milestone_inactive_to_active(self):
                 self.test_milestone
             )
 
+    def test_add_course_content_milestone_invalid_requirements(self):
+        """ Unit Test: test_add_course_content_milestone_invalid_requirements """
+        with self.assertRaises(exceptions.InvalidRequirementsException):
+            api.add_course_content_milestone(
+                self.test_course_key,
+                self.test_content_key,
+                self.relationship_types['REQUIRES'],
+                self.test_milestone,
+                set()  # Not JSON serializable
+            )
+
+    def test_add_course_content_milestone_active_exists_update_requirements(self):
+        """ Unit Test: test_add_course_content_milestone_active_exists_update_requirement """
+        api.add_course_content_milestone(
+            self.test_course_key,
+            self.test_content_key,
+            self.relationship_types['REQUIRES'],
+            self.test_milestone,
+            {'min_score': 60}
```

Comment:
Also, another thought -- what if we created a Requirements model and linked it back to CourseContentMilestone -- then we could have a 1:M thing and also link Requirements to other types of milestones, such as CourseMilestone.